test/functional/test_runner.py

```diff
@@ -234,6 +234,7 @@ def main():
     parser.add_argument('--quiet', '-q', action='store_true', help='only print dots, results summary and failure logs')
     parser.add_argument('--tmpdirprefix', '-t', default=tempfile.gettempdir(), help="Root directory for datadirs")
     parser.add_argument('--failfast', action='store_true', help='stop execution after the first test failure')
+    parser.add_argument('--filter', help='regular expression to filter scripts to run')
```

Comment:
Perhaps place it alphabetically, e.g. between extended and help. The failfast placement looks like an anomaly.

Suggested help: "filter tests to run by regular expression", or possibly "filter scripts to run by regular expression", in the format of verb + subject + details. FWIW I've rewritten the other argument help messages here and in test_framework.py for a PR following this format.

Reply:
Argument `--ci` is also misplaced. Note that `test/functional/test_runner.py --help` already has the output sorted.

Updated message.

Reply:
When I run `test/functional/test_runner.py --help` on Linux Debian, the output is not sorted (it's a bit of a mess TBH and am writing a PR to improve that and a few other things with the test runner output).